.github/workflows/semgrep.yaml

```diff
@@ -6,8 +6,21 @@ on:
     - master
 name: Semgrep
 jobs:
+  has-secrets:
+    name: Check Secrets
+    runs-on: ubuntu-latest
+    steps:
+    - id: semgrep
+      env:
+        SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
```

Comment:
I'm flexible. This appears to be the officially recommended approach. They more or less commit to trying not to leak a secret as long as it's used as a whole thing.

Because of the way it's written, while the secret _technically_ reaches the github runner, it's never used by a shell.

Reply:
I added a `HAS_SECRET_ACCESS` secret.

I'm probably just being paranoid, but I'd feel more comfortable if we used this.